### ISSUE TITLE
Supporting typefaces with ligatures

### ISSUE THREAD

**Issue by MarcusHallettUAH** [Microsoft/WSL#1288]
I gather some individuals<sup>[[who?]](#)</sup> desire the ability to use typefaces with "programming" or "console" ligatures in the console.

To my knowledge conhost doesn't support ligatures of any form.<sup>[1](#MHNote1)</sup> Would this be worthwhile?
###### See Also

_I think this is at least tangentially related to the following:_
Can't set different font for the terminal #757 
Use a fallback font to display missing characters #1124 
Feature requests #1099 
Consider adding padding and line height options #245 
Terminal ought to work with more fonts #405 

I'll drop this on the console User Voice. I think it's tangentially related to Bash on Windows, so I'm raising this here. This is a possible duplicate of #405; however, I believe it has a narrower scope than that question, meriting a unique issue. 
###### Notes

<a id="MHNote1">1</a>: As of build 14955.

**Comment by fpqc** [comment]
@MarcusHallettUAH Heya, I have it on good authority that any OTF support at all is pretty low priority on the backlog. I'm guessing that this ask is even lower a fortiori.

**Comment by zadjii-msft** [comment]
@MarcusHallettUAH I appreciate the hell out of what has got to be the best formatted issue report I've seen. 

That being said, yea this is a dupe of #405. We have recently received some specific impulse to try and pursue this feature, so it's moved up in the backlog, but it's still blocked by dependency on other features that aren't going to make it in this release.

**Comment by bitcrazed** [comment]
Thanks for the feedback. 

I am a big fan of FiraCode and the like and look forward to the Console one day supporting ligature monospace fonts, etc., but it is dependent on us building a new text renderer based on DirectWrite vs. our current GDI-based renderer.

This is a work item that's starting to get quite close to the top of our priority list, but it's a significant work item, so it'll take a little while to complete. 

HOWEVER, we WILL make A LOT of noise when this does happen, so I'd be surprised if you miss it when we do announce ;)

And as @zadjii-msft says, **GREAT JOB** @MarcusHallettUAH for writing a clear, concise bug 🥇 

Closing for now as this is in the WSL repo, a dupe, and already on our work list.
